### PR TITLE
Consolidate gstreamer vpp metrics

### PR DESCRIPTION
**Pull request by uartie**
Added vpp metric mixin to consolidate/share code with all gstreamer tests.  FFMpeg tests will be consolidated later.

### CODE REVIEW

File: lib/mixin/vpp.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
+###
+### Copyright (C) 2018-2021 Intel Corporation
```
[the remark this answers is not in the document]
Comment (uartie) [reply lib/mixin/vpp.py:2]: Most of the code was developed before 2021, just moved here.

File: lib/mixin/__init__.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+###
+### Copyright (C) 2018-2021 Intel Corporation
```
[the remark this answers is not in the document]
Comment (uartie) [reply lib/mixin/__init__.py:2]: Yep, will fix.

File: lib/mixin/vpp.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      assert abs(ref[-3] - actual[-3]) <  0.2, "Luma (Y) out of baseline range"
+
+    get_media().baseline.check_result(
+      compare = compare, context = self.refctx, psnr = psnr)
+
+  def check_default(self):
+    metrics.check_metric(**vars(self))
+
+  def check_metrics(self):
+    getattr(self, "check_{vpp_op}".format(**vars(self)), self.check_default)()
```
[the remark this answers is not in the document]
Comment (uartie) [reply lib/mixin/vpp.py:144]: If subclass does not define vpp_op, then it is a developer bug and we should let it break so we know it needs fixed.

File: test/gst-msdk/vpp/brightness.py [hunk 4]
```diff
@@ -15,9 +15,8 @@
 class default(VppTest):
   def before(self):
     vars(self).update(
-      caps        = platform.get_caps("vpp", "brightness"),
-      vpp_op = "brightness",
-      NOOP        = 50 # i.e. 0.0 in msdkvpp range should result in no-op result
```
[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/brightness.py:20]: The NOOP logic was normalized in https://github.com/intel/vaapi-fits/pull/303.  Now, level 50 for procamp in config is always a no-op regardless of middleware default.

[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/brightness.py:20]: So there is no behavior change.

File: test/gst-msdk/vpp/contrast.py [hunk 5]
```diff
@@ -15,9 +15,8 @@
 class default(VppTest):
   def before(self):
     vars(self).update(
-      caps        = platform.get_caps("vpp", "contrast"),
-      vpp_op = "contrast",
-      NOOP        = 50 # i.e. 1.0 in msdkvpp range should result in no-op result
```
[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/contrast.py:20]: no-op means "no operation".  That is, at this level (50) the plugin should just passthrough the input without any changes.

File: lib/mixin/vpp.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (uartie) [reply lib/mixin/vpp.py:144]: That is, all subclasses *must* define a vpp_op property.

File: test/gst-msdk/vpp/brightness.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/brightness.py:20]: ... and no-op (NOOP) is fixed based on new mapping rules.

File: test/gst-msdk/vpp/contrast.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/contrast.py:20]: The vpp mixin checks no-op logic.

File: lib/mixin/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (uartie) [reply lib/mixin/__init__.py]: done

File: test/gst-msdk/vpp/brightness.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (uartie) [reply test/gst-msdk/vpp/brightness.py:20]: specifically, level 50 is always mapped to the middleware default since #303.

**Review by FocusLuo — APPROVED** [review]
LGTM,
Only feel that It seems like there are too many commit in a PR

**Comment by uartie** [comment]
> LGTM,
> Only feel that It seems like there are too many commit in a PR

Would you prefer we squash some?

**Comment by FocusLuo** [comment]
> > LGTM,
> > Only feel that It seems like there are too many commit in a PR
> 
> Would you prefer we squash some?
If it is possible, maybe 10-15 commits will be better.

Will there be also a ffmpeg consolidate gstreamer vpp metrics PR later?

**Comment by uartie** [comment]
I will squash lib/mixin commits into one and test/<middleware> commits into others.

> > > LGTM,
> > > Only feel that It seems like there are too many commit in a PR
> > 
> > 
> > Would you prefer we squash some?
>  If it is possible, maybe ~15 commits will be better.
> 

Done. Squashed to 6 commits

> Will there be also a ffmpeg consolidate vpp metrics PR later?

Yes, I will work on ffmpeg consolidation after gstreamer consolidation is finished.

**Pull request merged**